src/components/Tag/Tag.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { useTheme } from '../../theme'
+import { GU, textStyle } from '../../style'
+
+const NORMAL = 'normal'
+const COMPACT = 'compact'
+const INFO = 'info'
+const WARNING = 'warning'
+const HELP = 'help'
+const TAG = 'tag'
+const APP = 'app'
+const IDENTITY = 'identity'
+const NOTIFICATION = 'notification'
+
+function useMode(mode) {
+  const theme = useTheme()
+
+  if (mode === WARNING) {
+    return `
+      background: ${theme.warningSurface};
+      color: ${theme.warningSurfaceContent};
+    `
+  }
+
+  if (mode === HELP) {
+    return `
+      background: ${theme.helpSurface};
+      color: ${theme.helpSurfaceContent};
+    `
+  }
+
+  if (mode === TAG) {
+    return `
+      background: ${theme.tag};
+      color: ${theme.tagContent};
+    `
+  }
+
+  if (mode === APP) {
+  }
+
+  if (mode === IDENTITY) {
+  }
+
+  if (mode === NOTIFICATION) {
+  }
+
+  // info
+  return `
+    background: ${theme.infoSurface.alpha(0.08)};
+    color: ${theme.infoSurfaceContent};
+  `
+}
+
+function useSize(size) {
+  if (size === COMPACT) {
+    return `
+      ${textStyle('label3')};
+      padding: 0 ${0.5 * GU}px;
+      min-width: 14px;
+    `
+  }
+
+  // normal
+  return `
+    ${textStyle('label2')};
+    padding: 0 ${1.5 * GU}px;
+    min-width: 22px;
+  `
+}
+
+function Tag({ children, mode, size, uppercase, color, background, ...props }) {
+  const modeStyles = useMode(mode)
+  const sizeStyles = useSize(size)
+
+  return (
+    <span
+      css={`
+        display: inline-flex;
+        align-items: center;
+        justify-content: center;
+        border-radius: 100px;
+        white-space: nowrap;
```

Comment:
Should we add a fixed height, since it’s only using one line anyway? At the moment the height is 22 and 15, while it should be 24 and 16 (from the Figma file).

Also could we add, in normal mode and when `uppercase` is on, a padding-top of `1.5px` to adjust the font alignment? Something like this (except you might want to have these in `useSize()`):

```suggestion
        white-space: nowrap;
        height: ${size === NORMAL? 3 * GU : 2 * GU}px;
        padding-top: ${size === NORMAL && uppercase? '1.5px' : '0'};
```